### PR TITLE
Introduces the hooks listener

### PR DESCRIPTION
**Pull request by anthonybishopric**
This is the draft of the thing that will listen to Consul for hooks as they are added to the system.

More tests are needed / integration with preparer is coming soon

### CODE REVIEW

File: bin/p2-bootstrap/bootstrap.go [hunk 1]
```diff
@@ -37,6 +37,7 @@ func main() {
 			log.Fatalf("Could not get consul manifest: %s", err)
 		}
 		consulPod := pods.PodFromManifestId(consulManifest.ID())
+		consulPod.RunAs = "root"
```
Comment (anthonybishopric) [inline bin/p2-bootstrap/bootstrap.go]: I had to make the `RunAs` for a pod configurable, which had the nice effect of removing the hardcoded pod names that were present in the pod package.

File: bin/p2-watch/main.go [hunk 2]
```diff
@@ -35,13 +34,13 @@ func main() {
 
 	quit := make(chan struct{})
 	errChan := make(chan error)
-	podCh := make(chan pods.PodManifest)
+	podCh := make(chan intent.ManifestResult)
```
Comment (anthonybishopric) [inline bin/p2-watch/main.go]: I switched this to a wrapper object so I could see what the path in Consul was when we read the pod.

File: pkg/hooks/listener.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+	if err != nil {
+		return util.Errorf("Couldn't make event exec dir %s", eventExecDir)
+	}
+	launchables, err := hookPod.GetLaunchables(manifest)
+	if err != nil {
+		return err
+	}
+
+	// First remove any pre-existing hooks for that pod.
+	podHookPattern := path.Join(eventExecDir, fmt.Sprintf("%s__*", hookPod.Id))
```
Comment (anthonybishopric) [inline pkg/hooks/listener.go]: Here I'm kind of nastily globbing every symlink from the previous version of the pod and removing them. Notice two bad things - one that there isn't a better grouping mechanism (like a sub folder) for hooks created by a particular pod - and two, that I just happen to know that `{podId}__*` is the wildcard for the name of hooks assigned to a pod. 

To do: refactor the hooks library to execute hooks in directories recursively.

**Comment by tummychow** [comment]
Other than those comments, lgtm, but that commit was pretty big so I can't be sure I didn't miss anything

**Pull request merged**